qtawesome/iconic_font.py

```diff
@@ -247,17 +251,18 @@ def icon(self, *names, **kwargs):
             error = '"options" must be a list of size {0}'.format(len(names))
             raise Exception(error)
 
-        parsed_options = []
-        for i in range(len(options_list)):
-            specific_options = options_list[i]
-            parsed_options.append(self._parse_options(specific_options,
-                                                      general_options,
-                                                      names[i]))
+        if QApplication.instance() is not None:
+            parsed_options = []
+            for i in range(len(options_list)):
+                specific_options = options_list[i]
+                parsed_options.append(self._parse_options(specific_options,
+                                                          general_options,
+                                                          names[i]))
 
-        # Process high level API
-        api_options = parsed_options
+            # Process high level API
+            api_options = parsed_options
 
-        return self._icon_by_painter(self.painter, api_options)
+            return self._icon_by_painter(self.painter, api_options)
```

Comment:
Maybe the else should return an empty QIcon plus some [warning](https://docs.python.org/2/library/warnings.html#warnings.warn)?

```
else:
   warning.warn('You need to have a running QApplication to use QtAwesome!')
   return QIcon()
```